### ISSUE TITLE
Bug report should pre-populate environment information

### ISSUE THREAD

**Issue by tlhackque** [sqlitebrowser/sqlitebrowser#1497]
Describe the new feature
--------------------------
Bug Report opens the GitHub issues page with the default web browser.

It should collect the environment information (client OS, DB4S version & variant) first and either

- Pre-populate the issues form; or
- Stick it in the paste buffer (with an indication to the user) so that it's a simple "paste" to fill it in

This would be more convenient for the user, and give you more accurate/complete data.  E.g., you can easily get the Windows Build or Linux uname -a strings, and your own commit ID.  And the versions of SQLite, DLL/.so, compiler, etc.

Additional info
---------------
Please answer these questions before submitting your feature request.

#### Is your feature request related to an issue? Please include the issue number.
Vaguely to #1434

#### Does this feature exist in another product or project? Please provide a link.

Several other products provide a mailto link that pre-populates subject & body.

Perl provides the perlbug script, which prepares and sends a mail message and includes detailed environment information.

**Comment by justinclift** [comment]
Hmmm, isn't this somewhat implemented in #1388?

**Comment by tlhackque** [comment]
Yes, somewhat -  I got stopped at:

> Please note that GitHub no longer supports Internet Explorer.
> We recommend upgrading to the latest Microsoft Edge, Google Chrome, or Firefox.

(IE isn't my favorite or even most frequent, but is the default due to some inflexible wet xxx soft ware.  Which is why I prefer maiilto - but that's another story.  I guess GitHub is pushing Edge since it got bought by Microsoft.)

Slogging on, I get the one line:
> DB4S v3.10.99 (Jul 30 2018) on Windows 10 (10.0) (winnt/10.0.17134) [x86_64]

I haven't touched the DB4S crypt version - hopefully the one line would be different...

But as noted in 1388, the issues form (template) is lost.  So perhaps this should be recast as "enhance to use the paste buffer instead, and collect more detail"  The model is

[Your environment - Please paste details from DB4S Help->Bug Report]
...

You might have "Copy (or Show) environment" as well as "Go to website" as options.

I would want to know compiler, its options, libraries, and commit - as I've been bitten by these.  More of a Unix issue than Windows, though you will see the odd Windows user who decides to build with ICC/CLang and/or some unexpected version of the database.  Or load the wrong DLL.

For a comprehensive example of what's possible (and useful), look at perl -V - and also, see git's export-subst and ident (in .gitattributes)  It usually takes a mix of help from the build system and preprocessor to collect it all.

It is unreasonable to ask a user to enter this level of detail - but straightforward to automate.

**Comment by mgrojo** [comment]
I get this with "Bug Report":
> DB4S v3.10.99 (Aug  9 2018) on Ubunt 16.04.5 LTS (linux/4.4.0-131-generic) [x86_64]

While in the About window I have:

> Version 3.10.99 (Aug  9 2018)
> 
> Qt Version 5.5.1
> 
> SQLCipher Version 3.8.6

So we should at least add those last two. In fact, both the About data and the environment information for bug reports should be the same. In that window the information can be copied, so not sure if we would need then a "Copy (or Show) environment" option.

**Comment by karim** [comment]
@tlhackque 

The old behavior was to open the link to the issues page, and let the user create the issue and fill in all the details. This was changed to use GitHub query parameters to do that automatically, so this was considered an improvement back then. :smiley:

What we required with every bug report was DB4S version and the user platform to know the affected version and try to reproduce it, or see if it was already fixed in another version. Most of the time this was enough and if we needed more info we would ask. :smiley: 

Having the full user system information is a good idea but is not going to work the way it is now. GitHub query parameters must be sent using GET, as you probably saw in the URL after clicking Bug Report.

> I guess GitHub is pushing Edge since it got bought by Microsoft.

GitHub was doing that before the Microsoft deal. I was getting the same message, and not just on GitHub. I guess because Internet Explorer is deprecated now.

> I haven't touched the DB4S crypt version - hopefully the one line would be different...

Unfortunately Qt and SQLite/SQLCipher versions are not included.

@mgrojo 

> So we should at least add those last two.

I fully agree. Qt and SQLite/SQLCipher should be added. I guess I didn't include them cause I was trying to replicate the manual bug report and filling in some info for the user which only included DB4S version and the platform.

**Comment by mgrojo** [comment]
I get this:
> DB4S v3.10.99 (Aug 10 2018) on Ubuntu 16.04.5 LTS (linux/4.4.0-131-generic) [x86_64]
> using SQLite Version 3.8.6
> and Qt 5.5.1

**Comment by mgrojo** [comment]
I'm not getting the SQLCipher version, but I built with 
`cmake -Dsqlcipher=1 .`
make
And the "Set Encryption" menu option is visible.

**Comment by MKleusberg** [comment]
Hmmm, can you step through the DBBrowserDB::getSqliteVersion() function and see were it fails? And does ```PRAGMA cipher_version;``` work for you when you run that in the Execute SQL tab?

**Comment by mgrojo** [comment]
It seems that my build does not actually use SQLcipher, although it has been requested and the menu options are indicating that.
- PRAGMA cipher_version returns nothing.
- ldd of sqlitebrowser executable shows libsqlite3.so and not libsqlcipher.so.
- And finally if I actually try do encrypt a database I get this error:
`no such function: sqlcipher_export (SELECT sqlcipher_export('sqlitebrowser_edit_encryption');)`

My installed packages include:
```
libsqlcipher-dev/xenial,now 3.2.0-1.1 amd64 [instalado]
  SQLCipher development files

libsqlcipher0/xenial,now 3.2.0-1.1 amd64 [instalado, automático]
  SQLCipher shared library

sqlcipher/xenial,now 3.2.0-1.1 amd64 [instalado]
  Command line interface for SQLCipher
```
I don't understand how my build can succeed but the sqlcipher library is not included.

**Comment by MKleusberg** [comment]
That's weird. I mean it makes sense that you can compile DB4S with encryption enabled but link it against SQLite instead of SQLCipher because the two share the same interface. But of course this isn't supposed to happen and looking at the CMakeLists.txt file I don't see any way this could happen either.

What happens if you build with qmake? Like this (or something similar):
```
qmake sqlitebrowser.pro CONFIG+=sqlcipher
make
```

**Comment by mgrojo** [comment]
I'm unable to compile using qmake. Using your command line it ends without a visible error, but it doesn't link an executable.

After some tests, I've finally get a correct compilation. It seems to be a problem of a clean state. With "git clean -f" wasn't enough, I had to remove manually all files, then checkout master and then revert all the deletions. There is probably an easier way.

Now I get this:
> SQLCipher Version 3.2.0 (based on SQLite 3.8.6)

And the encryption is now working fine.

**Comment by mgrojo** [comment]
I think the questions that we are using in the template are useful for helping users to open clear bug reports:
 
> #### What did you do?
> 
> 
> #### What did you expect to see?
> 
> 
> #### What did you see instead?

The template cannot be used, but I think we should copy them to the body.

**Comment by justinclift** [comment]
> With "make clean" and "git clean -f" wasn't enough

I tend to use `git clean -dffx` and `git reset --hard HEAD`, which resets the git state to the same as a newly checked out branch + deletes any unknown files/dirs too.

Can be useful to throw it in a shell alias, like:

```
alias regit=`git clean -dffx; git reset --hard HEAD`
```

So then just typing `regit` makes for a fresh repo. :smile:

**Comment by MKleusberg** [comment]
But that's just the bug report template, isn't it? I think for feature request or questions we have a different one. The menu item is named "Bug Report" to be fair though but if it's a feature request you have you'd still click on that menu item :wink:

**Comment by mgrojo** [comment]
> But that's just the bug report template, isn't it? I think for feature request or questions we have a different one. The menu item is named "Bug Report" to be fair though but if it's a feature request you have you'd still click on that menu item :wink:

Then we should add a different menu entry for reporting a feature request and give a different body for each entry.

**Comment by karim** [comment]
@mgrojo You don't need to copy the whole text to the body since we don't need to include any dynamic data to it. Just replace the `body` element with `template=Feature_request.md` instead. The final URL will look like this https://github.com/sqlitebrowser/sqlitebrowser/issues/new?labels=enhancement&template=Feature_request.md and give the exact same result.

**Comment by mgrojo** [comment]
@karim, nice. I didn't know that. Would it also work for bug reports which need to include the version information in the body?

**Comment by karim** [comment]
@mgrojo No, It won't work. `template` and `body` are mutually exclusive. We can use one or the other but not both (more info in #1388 comments).

If we want to include anything in the issue then we must use the `body` element, like in bug report. For feature request we can use any, but I prefer `template` since it creates a shorter URL.

**Comment by mgrojo** [comment]
Then we should use the `template` parameter unless we have to add something to the `body`, don't we?  I'll reimplement then the Feature Request action using the template and leave the bug action as is. Do you agree, @karim?

**Comment by karim** [comment]
> Then we should use the `template` parameter unless we have to add something to the `body`, don't we?

Yes, correct.

> I'll reimplement then the Feature Request action using the template and leave the bug action as is. Do you agree, @karim?

Yes, I think it is the best approach here.
